src/query/algorithms/triangles/Triangles.cpp

```diff
@@ -152,13 +148,15 @@ TriangleResult Triangles::countTriangles(map<long, unordered_set<long>> &central
                                 set2.insert(varThree);
                                 triangleCount++;
                                 if (returnTriangles) {
+                                    // TODO(thevindu-w): Flush to file on count exceeds value
```

Comment:
Lets get this fixed so that we can enhance the reliability.

Reply:
This is related to memory optimization. It should be a separate PR.

Reply:
Ok, hope you have code ready for the next PR you mentioned.